### PR TITLE
Add readable duration util

### PR DESCRIPTION
**Pull request by samjhayes**
**What does this Pull Request do?**
Add a new `readable_duration` utility function. Pass this a `datetime.timedelta` and it will return a human-readable duration string (for example "1 hour 15 minutes").

**Are there new tests?**
Yep

**Any background context you want to provide?**
I needed this for the new Metaverse schedule (which uses the Schedule modelling in Brockman). This util is based on a function originally written for Metaverse GrowTix schedules – I've moved it here and made a few improvements / added tests. I've also replaced the original function with this util (in an upcoming PR).

### CODE REVIEW

File: gn_django/utils/general.py [hunk 1]
```diff
@@ -48,3 +50,32 @@ def is_sphinx_autodoc_running():
...
+    """
+    Utility to return a human-readable duration string from a timedelta.
+
+    Args:
+        * `duration` - `timedelta`
+
+    Returns:
+        * `string`
+    """
+    if not duration or not isinstance(duration, timedelta):
```
Comment (deains) [inline gn_django/utils/general.py:65]: `not timedelta(seconds=0) == True`, you might want to have something that handles that.

File: gn_django/utils/general.py [hunk 2]
```diff
@@ -48,3 +50,32 @@ def is_sphinx_autodoc_running():
...
+        return ''
+
+    days = duration.days
+    seconds = duration.seconds
+    hours = floor(seconds / 3600)
+    minutes = floor((seconds - (hours * 3600)) / 60)
```
Comment (deains) [inline gn_django/utils/general.py:71]: `divmod` makes this nicer:
```
days, remainder = divmod(delta.total_seconds(), 86400)
hours, remainder = divmod(remainder, 3600)
minutes, seconds = divmod(remainder, 60)
```

Or alternatively:
```
days = duration.days
hours, remainder = divmod(duration.seconds, 3600)
minutes, seconds = divmod(remainder, 60)
```

**Review by deains — APPROVED** [review]
This is fine as is although you might want to make some additions to harden this a bit. Currently:

* This won't support `timedelta(seconds=0)` properly.
* Negative deltas like `timedelta(seconds=-600)` will produce very unexpected results!
* This will return "0 minutes" for deltas under 60 seconds.

Have a look at my timer formatter for racetime.gg for some inspiration. 🏁 https://github.com/racetimeGG/racetime-app/blob/master/racetime/utils.py#L494

File: gn_django/utils/general.py (hunk 2 above)
Comment (samjhayes) [reply gn_django/utils/general.py:71]: Ooh yeah that is nice

File: gn_django/utils/general.py (hunk 1 above)
Comment (samjhayes) [reply gn_django/utils/general.py:65]: I think that's probably ok - it will just return a blank string anyway?

Comment (samjhayes) [reply gn_django/utils/general.py:67]: I did it anyway https://github.com/gamernetwork/gn-django/pull/155/commits/1229f078bc2e68651e7ac7c720f9381c051d075b

File: gn_django/utils/general.py (hunk 2 above)
Comment (samjhayes) [reply gn_django/utils/general.py]: https://github.com/gamernetwork/gn-django/pull/155/commits/1229f078bc2e68651e7ac7c720f9381c051d075b

**Review by deains — APPROVED** [review]

**Pull request merged**